lib/PPI/Statement/Include.pm

```diff
@@ -270,6 +270,8 @@ sub feature_mods {
 		return { map +( $_ => $on_or_off ), @features } if @features;
 	}
 
+	return { try => 1 } if $self->module eq "Syntax::Keyword::Try";
```

Comment:
What about:
```
{ try => 'Syntax::Keyword::Try' }
```
and having possibility of multiple implementations, eg:
```
PPI::Statement::Compound::Try::Core
PPI::Statement::Compound::Try::Syntax::Keyword::Try
PPI::Statement::Compound::Try::Try::Tiny
...
```

Reply:
Good question. I would rather have try, try_keyword, try_tiny because these would only matter as different implementations of the parsing, and the parsing would query feature_mods for those keys to decide which path to take. This is a bit academic tho, as there currently is only one type of parsing implemented. Thoughts?

Reply:
I've spent some time thinking about it (in case of `try` even before it was added into core - shame on me, still didn't finish that project)

Implementation by class allows dynamic loading of class, so author of library (for example `Try::Tiny`) can provide rules for `PPI` to be able to handle its extension, whether it is DSL structure or keyword, as library uses it. It can be even standalone distribution.

Cons: such interface should be defined.

Example of another module which may benefit from such `PPI` capability
- `Syntax::Keyword::Match`

Here can be clash of my mindset vs not-my mindset.

I simply believe that future of the programming language lies in DDD + DSL as a representation of business domain processes. To be able to provide static analysis tool for it (critic, tidy, dead code detection, code duplication detection, automated refactoring ...) PPI already provides decent backend, but its extensibility is, sorry for rough word, bad.

As for example I'd share extension I'd like to prepare as PR:
replace snippet in `Token::Number` around line with:
```
set_class( 'Number::Octal' )
```

with (little bit abstract):
```
my @number_classes = (
  'PPI::Token::Number::Decimal',
  'PPI::Token::Number::Hex',
  ...
);

for my $class (@number_classes) {
  next unless $token->{content} =~ $class->regexp;
  $t->{token}->set_class( $class->as_set_class );
)
```

Reply:
Thanks for explaining, that actually makes more sense now, and i will make the feature_mods hash indicate the provider of the feature instead of simply a `1`. It costs relatively little effort and i understand how it opens doors for extensibility.

For the moment i will however not create sub-compounds, simply because it would be a lot of effort right now; but i am definitely open to doing it in the future when and if you would like to actually write code that makes use of those class names. :)

Reply:
I've updated the branch with this change.

Reply:
thanks, that looks great